RM-36943 Added annotations to the action output

janssen-tiobe opened the pull request.

janssen-tiobe committed:
#36943: Changed when annotations are being processed into messages in preparation of outputting json

janssen-tiobe committed:
#36943: Added function to create and set the `annotations` output variable

janssen-tiobe committed:
#36943: Wrote tests for action outputting, refactored a bit

Also made tests run way faster by disabling type checking when running tests

janssen-tiobe committed:
Merge branch 'main' into 36943-output_annotations_json

janssen-tiobe committed:
#36943: Added some usage documentation to the README

janssen-tiobe committed:
#36943: Fixed Coding Standard issue

janssen-tiobe committed:
#36943: reviewComments are now collected into a single group of annotations

Also fetching and processing annotations at once and fetching them always to put them into the output.

brouwers-tiobe commented on src/action/decorate/summary.ts:
```diff
@@ -47,8 +38,9 @@ export async function createSummaryBody(analysisResult: AnalysisResult): Promise
 
-      if (projectResult.reviewComments && projectResult.reviewComments.unpostable.length > 0) {
-        summary.addRaw(createUnpostableAnnotationsDetails(projectResult.reviewComments.unpostable));
+      const unpostableReviewComments = projectResult.annotations.filter(r => !r.postable);
```
Change name to unpostableAnnotations

brouwers-tiobe commented on src/analysis/client/analysis-results.ts:
```diff
@@ -14,12 +12,11 @@ import { getQualityGate, getQualityGateUrl } from '../../viewer/qualitygate';
  */
 export async function getClientAnalysisResults(explorerUrls: string[], changedFiles: ChangedFile[]): Promise<AnalysisResult> {
-  const hasExplorerUrl = explorerUrls.length !== 0;
   const analysisResult: AnalysisResult = {
```
Rather than creating analysisResult here and mutating it while iterating over the projectResults, I think it is more clear to first build the list of projectResults, then do all aggregation necessary to create analysisResult

janssen-tiobe committed:
#36943: Making getClientAnalysisResults more readable

by creating the analysis result when it is returned instead of creating it with dummy data that is later replaced.
